### PR TITLE
[jax2tf] Add a template file for documentation generation.

### PR DESCRIPTION
**Pull request by bchetioui**
The documentation now gives instructions about how to
regenerate it, as well as when it was last generated.

### CODE REVIEW

File: jax/experimental/jax2tf/tests/tf_test_util.py [hunk 1]
```diff
@@ -29,10 +29,13 @@
 
 import os
 
-if os.getenv('JAX2TF_CATEGORIZE_OUT'):
+if not os.getenv('JAX2TF_CATEGORIZE_OUT') is None:
```
Comment (gnecula) [inline jax/experimental/jax2tf/tests/tf_test_util.py:32]: I normally see this written as "x is not None"; feels more readable, wdyt?

File: jax/experimental/jax2tf/primitives_with_limited_support.md [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+*Last generated on (YYYY-MM-DD): 2020-09-08*
+
+## Updating the documentation
+
+This file is automatically generated by running the jax2tf tests in
+`jax/experimental/jax2tf/tests/primitives_test.py` with the
+`JAX2TF_CATEGORIZE_OUT` environment variable set.
+
+## Generated summary of primitives with limited support
+
```
Comment (gnecula) [inline jax/experimental/jax2tf/primitives_with_limited_support.md:12]: I suggest adding some text here: 

The following JAX primitives are converted to Tensorflow but the result of the conversion may trigger runtime errors when run on certain devices and with certain data types.

File: jax/experimental/jax2tf/primitives_with_limited_support.md [hunk 3]
```diff
@@ -29,3 +39,29 @@
 | scatter-mul | Missing TF support | scatter-mul is unimplemented for dtype complex64 | TPU |
 | select_and_gather_add | Missing TF support | select_and_gather_add is unimplemented for dtype float32 | TPU |
 | svd | Missing TF support | svd is unimplemented for dtype complex64; this works on JAX because JAX uses a custom implementation | CPU, GPU |
+
+## Not yet implemented primitive conversions
+
```
Comment (gnecula) [inline jax/experimental/jax2tf/primitives_with_limited_support.md:44]: I would add some text here as well:

The conversion of the following JAX primitives is not yet implemented. 

Nit: I would either put them all in one line, or put them in a table with similar format as the primitives above.

File: jax/experimental/jax2tf/tests/tf_test_util.py (hunk 1 above)
Comment (SIben) [reply jax/experimental/jax2tf/tests/tf_test_util.py]: Changed this in `bd4528a2`!

File: jax/experimental/jax2tf/primitives_with_limited_support.md (hunk 3 above)
Comment (SIben) [reply jax/experimental/jax2tf/primitives_with_limited_support.md:48]: Added & put all the primitives on one line.

**Pull request merged**